packages/opentelemetry-sdk-node/src/sdk.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { HttpTextPropagator, metrics } from '@opentelemetry/api';
+import { ContextManager } from '@opentelemetry/context-base';
+import { MeterConfig, MeterProvider } from '@opentelemetry/metrics';
+import { NodeTracerConfig, NodeTracerProvider } from '@opentelemetry/node';
+import { detectResources, Resource } from '@opentelemetry/resources';
+import { BatchSpanProcessor, SpanProcessor } from '@opentelemetry/tracing';
+import { NodeSDKConfiguration } from './types';
+
+/** This class represents everything needed to register a fully configured OpenTelemetry Node.js SDK */
+export class NodeSDK {
+  private _tracerProviderConfig?: {
+    tracerConfig: NodeTracerConfig;
+    spanProcessor: SpanProcessor;
+    contextManager?: ContextManager;
+    httpTextPropagator?: HttpTextPropagator;
+  };
+  private _meterProviderConfig?: MeterConfig;
+
+  private _resource: Resource;
+
+  private _autoDetectResources: boolean;
+
+  /**
+   * Create a new NodeJS SDK instance
+   */
+  public constructor(configuration: Partial<NodeSDKConfiguration> = {}) {
+    this._resource = configuration.resource ?? new Resource({});
+
+    this._autoDetectResources = configuration.autoDetectResources ?? true;
+
+    if (configuration.spanProcessor || configuration.traceExporter) {
+      const tracerProviderConfig = {} as NodeTracerConfig;
+
+      if (configuration.defaultAttributes) {
+        tracerProviderConfig.defaultAttributes =
+          configuration.defaultAttributes;
+      }
+      if (typeof configuration.logLevel === 'number') {
```

Comment:
except this what happen if you don't check for the existence ? not sure if the rest checks are needed or they should have some stronger type checks, WDYT ?

Reply:
Just because it could be `0` which would fail a simple if check. Did not want to use `== null` as some people don't like that and `=== undefined || === null` was too long.